opentelemetry-instrumentation/tests/test_auto_tracing.py

```diff
@@ -104,3 +125,24 @@ def test_trace_init_otlp(self):
             "my-otlp-test-service",
         )
         del environ["OTEL_SERVICE_NAME"]
+
+    @patch.dict(environ, {"OTEL_IDS_GENERATOR": "custom_ids_generator"})
+    @patch(
+        "opentelemetry.instrumentation.auto_instrumentation.components.trace.IdsGenerator",
+        new=IdsGenerator,
+    )
+    @patch(
+        "opentelemetry.instrumentation.auto_instrumentation.components.iter_entry_points"
+    )
+    def test_trace_init_custom_ids_generator(self, mock_iter_entry_points):
+        mock_iter_entry_points.configure_mock(
+            return_value=[
+                IterEntryPoint("custom_ids_generator", CustomIdsGenerator)
```

Comment:
I'm a little surprised you need this logic. I would have suggested we either ship a second IdsGenerator with the sdk so we can verify the IterEntrypoint, or ship one as part of the instrumentation.

This could break in the future since you're relying on an internal implementation detail, but the test is nice and small so it should be easy to catch.

Reply:
Thank you! 😄